### PR TITLE
Run typecheck on CI

### PR DESCRIPTION
**Pull request by angelikatyborska**
This would have caught https://github.com/bitcrowd/tickety-tick/issues/406

### CODE REVIEW

**Comment by klappradla** [comment]
Rebased this on `main` real quick 🏓

File: package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "lint": "eslint --ignore-path .gitignore --ext .js,.ts,.tsx .",
     "stylelint": "stylelint --ignore-path .gitignore 'src/**/*.scss'",
     "test": "jest",
+    "typecheck": "tsc --noEmit",
```
Comment (klappradla) [inline package.json:23]: 💯 

Does this actually mean that we can drop some stuff from ESLint?

File: tsconfig.json [hunk 2]
```diff
@@ -8,8 +8,9 @@
     "moduleResolution": "node",
     "noImplicitAny": true,
     "outDir": "./dist/",
+    "skipLibCheck": true,
```
Comment (klappradla) [inline tsconfig.json:11]: 👍 nice

**Review by klappradla — APPROVED** [review]
Great addition 🦄 

Wondering if we can slim down the ES-Lint stuff then 👀 ?

File: package.json (hunk 1 above)
Comment (angelikatyborska) [reply package.json:23]: The typescript eslint plugin still has some useful checks in there (this is the list of default rules in the "recommended" config that we're using: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/recommended.ts) so I would keep it

**Pull request merged**